nemo/src/io/parser.rs

```diff
@@ -690,6 +690,7 @@ impl<'a> RuleParser<'a> {
     /// predicate name together with its arity.
     fn parse_qualified_predicate_name(
         &'a self,
+        constraint_is_lower_bound: bool,
```

Comment:
When parsing source declarations, the type annotations are always lower bounds. In other cases like outputs, these requirements might be stricter. (Since import / export directives are WIP, I would not polish this right now as I expect things to change in the parser anyway.)